src/BasePagination.php

```diff
@@ -320,28 +328,18 @@ public function urlQueryParameters(array $value): static
      */
     protected function createUrl(int $page): string
     {
-        if ($this->urlGenerator === null) {
-            throw new Exception\UrlGeneratorNotSetException();
-        }
-
-        $pageConfig = $this->pageConfig;
-
-        if ($pageConfig === []) {
-            $pageConfig = [
-                $this->pageName => $page,
-                $this->pageSizeName => $this->getPaginator()->getPageSize(),
-            ];
-        }
-
-        $urlArguments = $this->urlArguments ?? [];
-        $urlQueryParameters = array_merge($pageConfig, $this->urlQueryParameters);
-
-        $urlName = $this->currentRoute->getName();
-
-        return match ($urlName) {
-            null => '?' . http_build_query($urlQueryParameters),
-            default => $this->urlGenerator->generate($urlName, $urlArguments, $urlQueryParameters),
-        };
+        return call_user_func(
+            $this->urlCreator ?? static fn(PageContext $context): string => '#' . $context->page,
+            new PageContext(
+                $page,
+                $this->getPaginator()->getPageSize(),
+                $this->pageParameterName,
+                $this->pageSizeParameterName,
+                $this->pageParameterPlace,
+                $this->pageSizeParameterPlace,
+                $this->queryParameters,
+            )
+        );
```

Comment:
This way we avoid extra call in default case:

```suggestion
        $context = new PageContext(
            $page,
            $this->getPaginator()->getPageSize(),
            $this->pageParameterName,
            $this->pageSizeParameterName,
            $this->pageParameterPlace,
            $this->pageSizeParameterPlace,
            $this->queryParameters,
        );

        if (is_callable($this->urlCreator)) {
            return $this->urlCreator($context);
        }
        
        return '#' . $context->page;
```

Reply:
done